File: Java/128.java
import java.util.*;

class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int i : nums) {
            set.add(i);
        }
        int max = 0;
        for (int i : nums) {
            if (!set.contains(i - 1)) {
                int currWinSize = 1;
                while (set.contains(i + 1)) {
                    set.remove(i + 1);
                    currWinSize++;
                    i++;
                }
                max = Math.max(max, currWinSize);
            }
        }
        return max;
    }
}

File: Python/128.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        result = 0
        numSet = set(nums)
        for num in numSet:
            if num - 1 not in numSet:
                length = 1
                while num + length in numSet:
                    length += 1
                result = max(result, length)

        return result
